src/screens/feed/useFeedModals.ts

```diff
@@ -0,0 +1,210 @@
+import { useAddPostWithNewId, useDeletePost } from 'dataAccess/mutations/useAddPost'
+import { useGetNotificationsConfig } from 'utils/notifications/notificationsConfig'
+import { useBooleanState } from 'hooks/useBooleanState'
+import { EditTargetType, FeedPost as FeedPostType } from 'mock-api/models/miragePostTypes'
+import { useState } from 'react'
+import { DrawerActions, useNavigation } from '@react-navigation/native'
+import { BottomTabNavigationType } from 'navigation/types'
+import { useTranslation } from 'react-i18next'
+import { useUserContext } from 'hooks/context-hooks/useUserContext'
+import EditIcon from 'assets/icons/icon-edit2.svg'
+import BinIcon from 'assets/icons/icon-bin.svg'
+import ReportIcon from 'assets/icons/circle-cross.svg'
+import HideIcon from 'assets/icons/icon-password-invisible.svg'
+import Svg from 'react-native-svg'
+import { Analytics } from 'services/analytics'
+
+type NavigationHookType = BottomTabNavigationType<'FEED'> & typeof DrawerActions
+
+type ModalOption = {
+  Icon: Svg
+  text: string
+  onPress: F0
+  show?: boolean
+}
+
+export const useFeedModals = (data: FeedPostType[] | undefined) => {
+  const [modalOptions, setModalOptions] = useState<ModalOption[]>([])
+  const [editTarget, setEditTarget] = useState<EditTargetType | null>()
+
+  const { mutate: deleteComment } = useDeleteComment()
+  const { mutate: editComment } = useEditComment()
+  const { mutate: deletePost } = useDeletePost()
+  const { mutate: addPostWithNewId } = useAddPostWithNewId()
+  const { user, updateUser } = useUserContext()
+
+  const { t } = useTranslation('feed')
+
+  const navigation = useNavigation<NavigationHookType>()
+
+  const [isMessageInputOpen, { setFalse: closeMessageInput, setTrue: openMessageInput }] =
+    useBooleanState(false)
+  const [isOptionsModalOpen, { setFalse: closeOptionsModal, setTrue: openOptionsModal }] =
+    useBooleanState(false)
+
+  const { notify } = useGetNotificationsConfig()
+
+  const handleSetMessageContent = (text: string) => {
+    if (editTarget?.type === 'comment') {
+      setEditTarget((prev) => prev && { ...prev, text })
+    }
+  }
+
+  const onPressModalDelete = (target: EditTargetType) => {
+    closeOptionsModal?.()
+    if (target?.type === 'comment') {
+      deleteComment(target.commentId, {
+        onSuccess: () => {
+          notify('successCustom', { params: { title: t('commentDeleted') } })
+        },
+      })
+    }
+    if (target?.type === 'post') {
+      const deletedPost = data?.find((post) => post.id === target.postId)
+      deletePost(target.postId, {
+        onSuccess: () => {
+          notify('successCustom', {
+            params: {
+              title: t('postDeleted'),
+              onPressText: t('undo'),
+              onPress: () => {
+                if (deletedPost) addPostWithNewId(deletedPost)
+              },
+            },
+          })
+        },
+      })
+    }
```

Comment:
Those two if statements could be extracted into its own funtions, and be called like
```typescript
if(target?.type === 'comment') handleCommentDeletion(target.commentId)
if(target?.type === 'post') handlePostDeletion(target.postId)
```
so it would be easier to see that the `onPressModalDelete`  function has to clear edit target and close the options modal besides deleting the content. WDYT?

Reply:
Not as a big deal if we were to implement the `onModalButtonPress` from my next comment :]